server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/SetMessages.feature

```diff
@@ -127,4 +127,26 @@ Feature: SetMessages method on shared folders
   Scenario: Draft creation in outbox is disabled
     Given "bob@domain.tld" has a mailbox "Outbox"
     When "bob@domain.tld" creates a draft message "mDraft" in mailbox "Outbox"
-    Then message "mDraft" is not created
\ No newline at end of file
+    Then message "mDraft" is not created
+
+  Scenario: A user can not move draft out of draft mailbox
+    Given "bob@domain.tld" has a mailbox "Drafts"
+    And "bob@domain.tld" creates a draft message "mDraft" in mailbox "Drafts"
+    When "bob@domain.tld" moves "mDraft" to user mailbox "shared"
+    Then message "mDraft" is not updated
+
+  Scenario: A user can not copy draft out of draft mailbox
+    Given "bob@domain.tld" has a mailbox "Drafts"
+    And "bob@domain.tld" creates a draft message "mDraft" in mailbox "Drafts"
+    When "bob@domain.tld" copies "mDraft" from mailbox "Drafts" to mailbox "shared"
+    Then message "mDraft" is not updated
+
+  Scenario: A user can not copy draft out of draft mailbox
+    Given "bob@domain.tld" has a mailbox "Drafts"
+    When "bob@domain.tld" moves "mBob" to user mailbox "Drafts"
+    Then message "mBob" is not updated
+
+  Scenario: A user can not copy draft out of draft mailbox
+    Given "bob@domain.tld" has a mailbox "Drafts"
+    When "bob@domain.tld" copies "mBob" from mailbox "shared" to mailbox "Drafts"
+    Then message "mBob" is not updated
```

Comment:
what happens if a user move a draft and at the same time remove the flag? It should be allowed, no?

Reply:
I will carft this out.